tools/aie2xclbin/XCLBinGen.cpp

```diff
@@ -501,6 +502,184 @@ static LogicalResult generateXCLBin(MLIRContext *context, ModuleOp moduleOp,
   return success();
 }
 
+static std::string chesshack(const std::string &input) {
```

Comment:
Can you factor this out into a library that can be called from aiecc.py?

Reply:
Just this one function or the whole thing?

Reply:
We shouldn't have the same code duplicated between aiecc.py and aie2xclbin.  This function seems like a good granularity to reduce that duplication?  I think the same goes for much of the other code you've factored out.